### PR TITLE
Fuse unmount

### PR DESCRIPTION
**Pull request by jinmouil**
Depends #16 

See the commit messages for details.

### CODE REVIEW

File: pkg/sentry/fsimpl/fuse/connection_control.go [hunk 1]
```diff
@@ -181,3 +183,63 @@ func (conn *connection) initProcessReply(out *linux.FUSEInitOut, hasSysAdminCap
...
+	for unique := range conn.fd.completions {
+		terminate = append(terminate, unique)
+	}
+
+	// Release all locks to avoid deadlock.
+	conn.asyncMu.Unlock()
+	conn.mu.Unlock()
+	conn.fd.mu.Unlock()
+
+	// 1. The requets blocked before initialization.
```
Comment (craig08) [inline pkg/sentry/fsimpl/fuse/connection_control.go:225]: request

File: pkg/sentry/fsimpl/fuse/dev.go [hunk 2]
```diff
@@ -343,7 +354,14 @@ func (fd *DeviceFD) Readiness(mask waiter.EventMask) waiter.EventMask {
...
 func (fd *DeviceFD) readinessLocked(mask waiter.EventMask) waiter.EventMask {
 	var ready waiter.EventMask
-	ready |= waiter.EventOut // FD is always writable
+
+	if fd.fs.umounted {
+		ready |= waiter.EventErr
+		return ready & mask
+	}
+
+	// FD is always writable
```
Comment (craig08) [inline pkg/sentry/fsimpl/fuse/dev.go]: Add period.

**Review by craig08 — APPROVED** [review]
LGTM

File: pkg/sentry/fsimpl/fuse/connection_control.go (hunk 1 above)
Comment (jinmouil) [reply pkg/sentry/fsimpl/fuse/connection_control.go:225]: There can be any number of requests.

**Pull request merged**